next-app/src/lib/server/sf.ts

```diff
@@ -1,18 +1,27 @@
 import { error } from '@sveltejs/kit'
+import type { HttpMethod } from '@sveltejs/kit/types/private'
 
-/**
- *
- * @typedef RPC
- * @type {object}
- * @property {string} name Name of the remote procedure to call
- * @property {string[]} [args] Arguments to pass to the remote procedure
- * @property {string} [cookie] https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie
- *
- * @param { RPC } rpc
- */
-export async function sf(rpc) {
-	const { name, args = [], cookie } = rpc
+export type Rpc = {
```

Comment:
I would appreciate something like this:
```
type SfEndpoint = 'set_project' | 'lext_stats' | 'set_project';

export type Rpc = {
  name: SfEndpoint,
  args?: string[] | object[],
  cookie?: string,
}
```

Reply:
On a more extreme note, if it tickles your fancy, you could also rewrite the `sf` method with overloads that define the return type based on the called endpoint:
```
export async function sf(name: 'activity_list_dto_for_current_project', { args = [], cookie = '' }: Rpc): Promise<LegacyResult>
export async function sf(name: SfEndpoint, { args = [], cookie = '' }: Rpc): Promise<any> {
  // ...
}
```

Reply:
I can appreciate where you're going there.  I personally prefer those incoming/outgoing types reside with their concrete implementations rather than in something I intended to be "generic", i.e., `sf.ts`, maybe there's a way to achieve both with that `&` operator perhaps?  I'll see where I get trying to achieve some compromise

Reply:
As I look at `next-app/src/lib/server/user.ts` I see that the encapsulation of the API contract is achieved programmatically as opposed to conventionally or via TS.  Since this is a bit more stylistic and nuanced, I'll proceed as is but am open to further discussions and possible changes in the future.

Reply:
@myieye you can actualy do event better and have a mapping type like this:
```ts
type mapper = {
'method1': ReturnType1,
'method2': ReturnType2
}
```
etc... it's a bit complicated to setup it's pretty cool. Though since sf will eventually go away I think it's overkill.